### PR TITLE
[Flutter Parent][MBL-14008] Custom help links

### PR DESCRIPTION
**Pull request by StephenBrough**
Testing:
 - My account (marlene) has a link to the canvas chat support as well as other types of help links.

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1026056&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1026057&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: apps/flutter_parent/test/screens/help/help_screen_test.dart [hunk 1]
```diff
@@ -41,20 +45,18 @@ void main() {
...
-
-    expect(find.text(l10n.help), findsOneWidget);
+  testWidgetsWithAccessibilityChecks('displays links', (tester) async {
+    HelpScreenInteractor interactor = _MockHelpScreenInteractor();
+    when(interactor.getObserverCustomHelpLinks(forceRefresh: anyNamed('forceRefresh')))
+        .thenAnswer((_) => Future.value([createHelpLink()]));
 
-    expect(find.text(l10n.helpSearchCanvasDocsLabel), findsOneWidget);
-    expect(find.text(l10n.helpSearchCanvasDocsDescription), findsOneWidget);
+    setupTestLocator((locator) => locator.registerLazySingleton<HelpScreenInteractor>(() => interactor));
```
Comment (CalvinKern) [inline apps/flutter_parent/test/screens/help/help_screen_test.dart]: This shouldn't be calling setupTestLocator inside individual tests, it's too messy and hard to maintain.

Instead, define the interactor by l10n, add it to the setupTestLocator called just in main (not in a test), and add implement the `setUp` method to reset the interactor (that way each test gets a clean interactor). See `assignment_details_interactor_test.dart` for an example if you need it.

File: apps/flutter_parent/lib/screens/help/help_screen.dart [hunk 2]
```diff
@@ -27,52 +29,102 @@ import 'package:intent/intent.dart' as android;
...
-        appBar: AppBar(
-          title: Text(l10n.help),
-          bottom: ParentTheme.of(context).appBarDivider(shadowInLightMode: false),
-        ),
-        body: ListView(
-          children: <Widget>[
-            ListTile(
-              title: Text(l10n.helpSearchCanvasDocsLabel),
-              subtitle: Text(l10n.helpSearchCanvasDocsDescription),
-              onTap: _showSearch,
```
Comment (CalvinKern) [inline apps/flutter_parent/lib/screens/help/help_screen.dart]: Can you delete all these unused strings from AppLocalizations?

File: apps/flutter_parent/lib/screens/help/help_screen.dart [hunk 3]
```diff
@@ -27,52 +29,102 @@ import 'package:intent/intent.dart' as android;
...
+      _handlePhoneLinks(link.url);
+    } else if (link.url.startsWith('mailto:')) {
+      // Support mailto links: https://community.canvaslms.com/docs/DOC-12664-4214610054
+      _handleMailtoLinks(link.url);
+    } else if (link.url.contains('cases.canvaslms.com/liveagentchat')) {
+      // Chat with Canvas Support - Doesn't seem work properly with WebViews, so we kick it out
+      // to the external browser
+      launch(link.url); // TODO: Make sure this works
+//      val intent = Intent(Intent.ACTION_VIEW).apply { data = Uri.parse(link.url) }
+//      startActivity(intent)
```
[the remark this answers is not in the document]
Comment (StephenBrough) [reply apps/flutter_parent/lib/screens/help/help_screen.dart]: lol, nope!

**Pull request merged**